py/surveyqa/summary.py

```diff
@@ -236,17 +232,60 @@ def last(arr):
         return arr[len(arr)-1]
     return exposures.group_by("TILEID").groups.aggregate(last)
 
-tzone = TimezoneInfo(utc_offset = -7*u.hour)
+utc_offset = -7*u.hour
+tzone = TimezoneInfo(utc_offset = utc_offset)
 t1 = Time(58821, format='mjd', scale='utc')
 t = t1.to_datetime(timezone=tzone)
 
-def get_surveyprogress(exposures, tiles, line_source, hover_follow, width=250, height=250, min_border_left=50, min_border_right=50):
+def get_progress(exposures, tiles, program):
+    '''Get survey and tile progress for a given program
+
+    Args:
+        exposures: Table with columns PROGRAM, TILEID, MJD
+        tiles: Table with columns TILEID, EXPOSEFAC
+        program: str program name to filter
+
+    Returns (time, survey_progress, tile_progress)
+
+    time = array of datetime objects;
+    survey_progress = array of EXPOSEFAC-weighted progress on observing tiles;
+    tile_progress = array of number of tiles observed
+    '''
+
+    keep = exposures['PROGRAM'] == program
+    exposures = exposures['TILEID', 'MJD'][keep]
+    keep = tiles['PROGRAM'] == program
+    tiles = tiles['TILEID', 'EXPOSEFAC'][keep]
+
+    #- Create table of last exposure for each tile, sorted by MJD
+    finished_tiles = exposures.group_by('TILEID').groups.aggregate(np.max)
+    finished_tiles = join(finished_tiles, tiles, keys='TILEID', join_type='left')
+    finished_tiles.sort('MJD')
+
+    #- Convert MJD to list of datetimes for bokeh
+    t1 = Time(finished_tiles['MJD'], format='mjd', scale='utc')
+    # t = t1.to_datetime(timezone=tzone)
+    # local non-timezone aware time, instead of much slower timezone aware
+    t = (t1 + utc_offset).to_datetime()
```

Comment:
Surprisingly, creating a timezone-aware `datetime` from `Time.to_datetime(timezone=tzinfo)` is much more expensive than creating a non-timezone aware `datetime`.  Since we just want an Arizona-local time datetime for plotting, offset the time by `utc_offset` (-7 hour) before creating the non-timezone aware datetime.

I would not have guess this, and only found it by profiling the code.